app/scripts/metamask-controller.js

```diff
@@ -1090,36 +1225,89 @@ export default class MetamaskController extends EventEmitter {
       initializeThreeBox: nodeify(this.initializeThreeBox, this),
 
       // permissions
-      approvePermissionsRequest: nodeify(
-        permissionsController.approvePermissionsRequest,
-        permissionsController,
+      clearPermissions: permissionController.clearState.bind(
```

Comment:
This API method is apparently unused! I have removed it in https://github.com/MetaMask/metamask-extension/pull/12731